### PR TITLE
vim-patch:8.2.4701: Kuka Robot Language files not recognized

### PR DESCRIPTION
**Pull request by clason**
Problem:    Kuka Robot Language files not recognized.
Solution:   Recognize *.src and *.dat files. (Patrick Meiser-Knosowski,
            closes vim/vim#10096)
https://github.com/vim/vim/commit/3ad2090316edc85e93094bba7af64f9991cc7f85

### CODE REVIEW

**Comment by clason** [comment]
Patterns like `*.src\c` aren't ported to `filetype.lua` yet -- is that a fundamental limitation? @gpanders

**Comment by gpanders** [comment]
Does the `\c` here mean case-insensitive?

If so, then we could include this in filetype.lua in the patterns section with something like `.*%.[Ss][Rr][Cc]` I think.

**Review by zeertzjq — COMMENTED** [review]
Missing `ft.vim`

**Comment by clason** [comment]
> Missing `ft.vim`

Oh, that's right, that file is vim9script now, and patches no longer apply cleanly...

**Comment by clason** [comment]
> Does the \c here mean case-insensitive?

I have no idea...

File: runtime/autoload/dist/ft.vim [hunk 1]
```diff
@@ -899,6 +899,23 @@ func dist#ft#FTtf()
   setf tf
 endfunc
 
+# Determine if a *.src file is Kuka Robot Language
```
Comment (zeertzjq) [inline runtime/autoload/dist/ft.vim:902]:
```suggestion
" Determine if a *.src file is Kuka Robot Language
```

File: runtime/autoload/dist/ft.vim [hunk 2]
```diff
@@ -899,6 +899,23 @@ func dist#ft#FTtf()
...
+func dist#ft#FTsrc()
+  if exists("g:filetype_src")
+    exe "setf " .. g:filetype_src
+  elseif getline(nextnonblank(1)) =~? '^\s*\%(&\w\+\|\%(global\s\+\)\?def\>\)'
+    setf krl
+  endif
+endfunc
+
+# Determine if a *.dat file is Kuka Robot Language
```
Comment (zeertzjq) [inline runtime/autoload/dist/ft.vim:911]:
```suggestion
" Determine if a *.dat file is Kuka Robot Language
```

Comment (clason) [reply runtime/autoload/dist/ft.vim:911]: 🤦

**Review by zeertzjq — APPROVED** [review]
`\c` means case-insensitive. There seems to be two ways to do this in Lua:
1. Convert the whole string to lower-case. However, `string.lower()` seems to be locale-dependent.
2. Just use `.*%.[Ss][Rr][Cc]`

**Review by zeertzjq — APPROVED** [review]

**Review by gpanders — APPROVED** [review]

**Pull request merged**